### DISCUSSION TITLE
Memory utilisation

### DISCUSSION THREAD

**Discussion by Manzs1288** [edgexfoundry/community#316] [Q&A] [answered]
Hi, 

Can someone help me how much memory(RAM and ROM) is required to run EdgeX foundry on custom hardware and X86 machine?

**Comment by cloudxxx8** [comment]
EdgeX Foundry is a micro service system, so it would depend on how many services you plan to run.
Here is the performance metrics data: https://lf-edgexfoundry.atlassian.net/wiki/spaces/FA/pages/11678627/Napa+Data

Generally, if you don't run the security service, 256 RAM RAM and 500 MB ROM is enough.
If you need the security services, 1 GB RAM and 1 GB ROM is enough.

**Reply by Manzs1288** [reply]
Thank you for response and sharing the link for metrics data.
I still have below queries, could you please help me with this:

1.  Statement "**Generally, if you don't run the security service, 256 RAM RAM and 500 MB ROM is enough.**"  does it mean that if I use all the service except security service then the required memory is  256 RAM RAM and 500 MB ROM?
2. In metrics data I see pi4 and dell Gateway as reference. I'm referring to non security sheets for both and I see the memory consumption for the same events it's different(For eg: for micro service "edgex-core-consul" the memory consumption is different)
3. For the actual memory consumption which section do I need to refer in metrics data I see different sectio.

**Reply by cloudxxx8** [reply, marked as answer]
1. it depends on what "all the services" are.  you need to choose the device service and app services you need.
2. the memory consumption from the real tests would always be floating. by the way, Consul is deprecated and will be replaced with a lighter home-made service core-keeper in the next release.
3. The Memory usage aggregation section is quite clear.

**Reply by Manzs1288** [reply]
Thank you it’s very clear now
